### ISSUE TITLE
Gem::ConfigMap is deprecated

### ISSUE THREAD

**Issue by GhaziTriki** [bigbluebutton/bigbluebutton#8455]
The recording process shows this

rap-archive-worker.rb[18202]: /usr/lib/ruby/vendor_ruby/rubygems/defaults/operating_system.rb:10: warning: constant Gem::ConfigMap is deprecated

rap-process-worker.rb[32653]: /usr/lib/ruby/vendor_ruby/rubygems/defaults/operating_system.rb:10: warning: constant Gem::ConfigMap is deprecated

Isn't it preferable to update this Gem?

**Comment by TheCellMC** [comment]
I'm also getting this issue

**Comment by eemirtekin** [comment]
I'm also getting this issue too.

**Comment by ffdixon** [comment]
The messages are warnings.  We're looking ahead to BigBlueButton 2.3 which will be on Ubuntu 18.04, so we'll fix it there with the newer version of ruby in 18.04.

**Comment by kunthar** [comment]
@ffdixon -- off topic-- would you consider registering with brave BAT programme? 
<img width="1403" alt="Screen Shot 2020-04-05 at 5 51 54 PM" src="https://user-images.githubusercontent.com/49767/78501836-5387ce00-7766-11ea-9780-f4772de33da3.png">

**Comment by isural** [comment]
This error is about gem version 3.1.x There are some suggestions to downgrade gem but i think for now it is not a big deal. As Fred mentioned, this will not brake BBB  working.

More details: https://github.com/rubygems/rubygems/issues/3068

**Comment by ffdixon** [comment]
@kunthar See http://docs.bigbluebutton.org/support/faq.html#how-can-i-donate-money-to-the-project.

**Comment by amg-web** [comment]
so if we add gem update --system 3.0.8 to installation script, it does not affect BBB ?
this error takes to much in logs and makes it fifficult to check for tother errors

**Comment by hex-m** [comment]
@ffdixon stated

>  We're looking ahead to BigBlueButton 2.3 which will be on Ubuntu 18.04, so we'll fix it there

So this issue was confirmed and thus should not be auto-closed.

**Comment by hex-m** [comment]
This seemingly got worse in `2.4.2`. I also get those warnings from the `sanity.rb`-script now.

```
Feb 02 14:58:37 bbb1 bbb-rap[30712]: Running sanity worker for xxxxxxxxxxxxxxxxxxxxxxxxxx-1642501692333
Feb 02 14:58:37 bbb1 bbb-rap[30712]: Task: Getting external meeting id
Feb 02 14:58:37 bbb1 bbb-rap[30712]: Task: Getting meeting metadata
Feb 02 14:58:37 bbb1 bbb-rap[30712]: Executing: ruby /usr/local/bigbluebutton/core/scripts/sanity/sanity.rb -m xxxxxxxxxxxxxxxxxxxxxxxxxx-1642501692333
Feb 02 14:58:37 bbb1 bbb-rap[30712]: /usr/lib/ruby/vendor_ruby/rubygems/defaults/operating_system.rb:10: warning: constant Gem::ConfigMap is deprecated
Feb 02 14:58:37 bbb1 bbb-rap[30712]: /usr/lib/ruby/vendor_ruby/rubygems/defaults/operating_system.rb:29: warning: constant Gem::ConfigMap is deprecated
Feb 02 14:58:37 bbb1 bbb-rap[30712]: /usr/lib/ruby/vendor_ruby/rubygems/defaults/operating_system.rb:30: warning: constant Gem::ConfigMap is deprecated
Feb 02 14:58:37 bbb1 bbb-rap[30712]: ruby: warning: shebang line ending with \r may cause problems
Feb 02 14:58:37 bbb1 bbb-rap[30712]: /usr/lib/ruby/vendor_ruby/rubygems/defaults/operating_system.rb:10: warning: constant Gem::ConfigMap is deprecated
Feb 02 14:58:38 bbb1 bbb-rap[30712]: Success?: true
```

**Comment by kepstin** [comment]
So, the fundamental problem here is that we're inappropriately installing system-wide 'bundler' gems in various places in the BBB packaging, such as in the [bbb-record-core](https://github.com/bigbluebutton/bigbluebutton/blob/8404048cfd35280d2dd427fc0c2e93790760c3f7/build/packages-template/bbb-record-core/after-install.sh#L42-L64), and possibly even forcibly upgrading the system 'rubygems' library, which we should not be doing.

We should instead be using the OS provided packages. The complication here is that Ubuntu 18.04 only provides bundler 1.16, so we will have to downgrade some of our existing Gemfile.lock files.

If we're making this sort of substantial change to the packaging, we should also fix the issue where we're installing ruby gems from the internet in the package post install script, those need to be bundled in the package instead.

**Comment by hex-m** [comment]
@antobinary mentioned in https://github.com/bigbluebutton/bigbluebutton/issues/12398#issuecomment-1009331854 that for `2.5` all the packaging-related parts should be in [the build directory](https://github.com/bigbluebutton/bigbluebutton/tree/develop/build). So substantial changes to the packaging are on their way it seems.

This issue should probably labeled with [deploy: packaging](https://github.com/bigbluebutton/bigbluebutton/issues?q=is%3Aopen+is%3Aissue+label%3A%22deploy%3A+packaging%22).

**Comment by antobinary** [comment]
> The complication here is that Ubuntu 18.04 only provides bundler 1.16, so we will have to downgrade some of our existing Gemfile.lock files.

Likely related: https://github.com/bigbluebutton/bigbluebutton/issues/14336

**Comment by antobinary** [comment]
Closely related to #14631 and possibly even full overlap

**Comment by antobinary** [comment]
Resolved after the upgrade to Ubuntu 20.04 and https://github.com/bigbluebutton/bigbluebutton/issues/14631
